lib/aur-chroot--standalone

```diff
@@ -0,0 +1,346 @@
+                           "$shrdir/makepkg.conf.d/$machine.conf")
+else
+    default_pacman_paths=("$etcdir/pacman-$machine.conf"
+                          "$shrdir/pacman.conf.d/aurutils-$machine.conf")
+
+    default_makepkg_paths=("$etcdir/makepkg-$machine.conf"
+                           "$shrdir/makepkg.conf.d/$machine.conf")
+fi
+
+# Change the default /usr/share/devtools/pacman-extra.conf in aur-chroot to
+# /etc/aurutils/pacman-<repo>.conf or /etc/aurutils/pacman-<uname>.conf in
+# aur-build, and pass it on to aur-chroot (#824, #846)
+for def in "${default_pacman_paths[@]}"; do
+    if [[ -f $def ]] && [[ ! -v pacman_conf ]]; then
+        pacman_conf=$def
+        break
+    fi
+done
+
+# The same as above but for /etc/aurutils/makepkg-<repo>.conf or
+# /etc/aurutils/makepkg-<uname>.conf. If the file is not found, fallback to
+# makepkg.conf files in /usr/share/devtools.
+for def in "${default_makepkg_paths[@]}"; do
+    if [[ -f $def ]] && [[ ! -v makepkg_conf ]]; then
+        makepkg_conf=$def
+        break
+    fi
+done
+
+# No pacman configuration is available for the container, or it points to a
+# non-existing file. Print a matching diagnostic and exit.
+if [[ ! -v pacman_conf ]]; then
+    diag_pacman_conf "${default_pacman_paths[@]}"
+    exit 2
+elif [[ ! -f $pacman_conf ]]; then
+    diag_pacman_conf "$pacman_conf"
+    exit 2
+elif [[ ! -v makepkg_conf ]]; then
+    diag_makepkg_conf "${default_makepkg_paths[@]}"
+    exit 2
+elif [[ ! -f $makepkg_conf ]]; then
+    diag_makepkg_conf "$makepkg_conf"
+    exit 2
+fi
+
+bindmounts_ro+=("$pacman_conf" "$makepkg_conf" /etc/pacman.d/mirrorlist)
+
+# Print paths to container and used makepkg/pacman paths. This does
+# not require a priorly created container.
+if (( status )); then
+    printf 'chroot:%s\npacman:%s\nmakepkg:%s\n' "$directory" "$pacman_conf" "$makepkg_conf"
+    exit 0
+fi
+
+is_multilib=0
+cache_dirs=()
+while read -r key _ value; do
+    case $key=$value in
+        Server=file://*)
+            # bind mount file:// paths to container (#461)
+            # required for update/build steps
+            bindmounts_rw+=("${value#file://}") ;;
+        CacheDir=*)
+            cache_dirs+=("${value}")
+            bindmounts_rw+=("${value}") ;;
+        \[multilib\]=)
+            is_multilib=1 ;;
+    esac
+done < <(pacman-conf --config "$pacman_conf")
+wait "$!"
+
+cleanup() {
+    rm "$directory"/root/aur-chroot--helper
+}
+
+trap cleanup EXIT
+
+# create new container, required for update/build steps
+if [[ ! -d $directory/root ]]; then
+    if (( !create )); then
+        printf >&2 '%s: %s is not a directory\n' "$argv0" "$directory"/root
+        printf >&2 '%s: did you run aur chroot --create?\n' "$argv0"
+        exit 20
+    fi
+
+    # default to base-devel or multilib-devel, unless packages are
+    # specified on the command-line.
+    # available packages depend on the configured pacman configuration
+    if (( $# )); then
+        base_packages=("$@")
```

Comment:
The the comment for `aur-build` on L129; this logic assumes makepkg arguments are specified with `--margs`, not specified after `--`.